tests/test_data_asset.py

```diff
@@ -1029,52 +1048,68 @@ def test_discard_failing_expectations(self):
         exp1 = [
             {'expectation_type': 'expect_column_to_exist',
              'kwargs': {'column': 'A'}},
-            {'expectation_type': 'expect_column_to_exist',
-             'kwargs': {'column': 'B'}},
-            {'expectation_type': 'expect_column_to_exist',
-             'kwargs': {'column': 'C'}},
-            {'expectation_type': 'expect_column_to_exist',
-             'kwargs': {'column': 'D'}},
             {'expectation_type': 'expect_column_values_to_be_in_set',
              'kwargs': {'column': 'A', 'value_set': [1, 2, 3, 4]}},
+            {'expectation_type': 'expect_column_to_exist',
+             'kwargs': {'column': 'B'}},
             {'expectation_type': 'expect_column_values_to_be_in_set',
              'kwargs': {'column': 'B', 'value_set': [5, 6, 7, 8]}},
+            {'expectation_type': 'expect_column_to_exist',
+             'kwargs': {'column': 'C'}},
             {'expectation_type': 'expect_column_values_to_be_in_set',
              'kwargs': {'column': 'C', 'value_set': ['a', 'b', 'c', 'd']}},
+            {'expectation_type': 'expect_column_to_exist',
+             'kwargs': {'column': 'D'}},
             {'expectation_type': 'expect_column_values_to_be_in_set',
              'kwargs': {'column': 'D', 'value_set': ['e', 'f', 'g', 'h']}}
         ]
 
         sub1 = df[:3]
 
         sub1.discard_failing_expectations()
-        self.assertEqual(sub1.find_expectations(), exp1)
+        # PY2 sorting is allowed and order not guaranteed
```

Comment:
Feels like sorting is a function that `ExpectationSuite.get_config` (or similar) should eventually handle.

Reply:
👍